django_cloud_tasks/models.py

```diff
@@ -0,0 +1,133 @@
+# pylint: disable=no-member
+from datetime import datetime
+from typing import Optional, Dict, List
+from django.db import transaction, models
+from django.db.models import QuerySet, Q
+from django.apps import apps
+from django_cloud_tasks import tasks, serializers
+
+
+class Pipeline(models.Model):
+    name = models.CharField(max_length=100)
+
+    def start(self):
+        routines = self.routines.filter(Q(dependent_routines__id__isnull=True) & ~Q(status=Routine.Statuses.COMPLETED))
+        for routine in routines:
+            routine.enqueue()
+
+    def revert(self):
+        routines = self.routines.filter(Q(next_routines__id__isnull=True) & ~Q(status=Routine.Statuses.REVERTED))
+        for routine in routines:
+            routine.revert()
+
+    def create_routines_and_vertex(self, routines: List[Dict]) -> QuerySet:
+        previous_routine = None
+        for routine_payload in routines:
+            routine = self.routines.create(**routine_payload)
+            if previous_routine:
+                RoutineVertex.objects.create(routine=previous_routine, next_routine=routine)
+            previous_routine = routine
+
+        return self.routines
```

Comment:
This method covers only **one** use case: a linear pipeline. Is it really useful?
Maybe helpers on Routine are more useful, such as `add_dependency` and `add_next`.

Reply:
For sure! This method is really useful. I want a helper to create routines with vertex, I believe it will cover most cases.

Reply:
I does not cover any case that's not linear.
A simple divide and merge cannot be created with this method.

Reply:
It covers most of our cases

Reply:
I just deleted this method